src/main/thrift/parquet.thrift

```diff
@@ -660,6 +660,22 @@ struct ColumnMetaData {
   13: optional list<PageEncodingStats> encoding_stats;
 }
 
+struct ColumnCryptoMetaData {
+  /** Path in schema **/
+  1: required list<string> path_in_schema
+  
+  /** Encrypted or plaintext **/
+  2: required bool encrypted
+  
+  /** Column can be encrypted with the footer key,
+   * or a column-specific key **/
+  3: required bool encrypted_with_footer_key
```

Comment:
I'd suggest using an union here too for encrypted_with_footer_key and column_key_metadata. How about a union like this:
```
union EncryptionMetadata {
 1: bool encrypted_with_footer_key
 2: binary column_key_metadata
}

```
or something like this:
```
union EncryptionMetadata {
 1: bool encrypted_with_footer_key
 2: ColumnKeyMetadata column_key_metadata
}

struct ColumnKeyMetadata {
  1: binary column_key_metadata
}

```

Reply:
I support Nándor's idea about using unions for mutually exclusive data fields, but I think the best way to do it would be as follows:

    union EncryptionMetadata {
      1: EncryptionWithFooterKey ENCRYPTION_WITH_FOOTER_KEY
      2: EncryptionWithColumnKey ENCRYPTION_WITH_COLUMN_KEY
    }
    
    struct EncryptionWithFooterKey {
    }
    
    struct EncryptionWithColumnKey {
      1: binary column_key_metadata
    }

Reply:
+1 for recommendation from @zivanfi

Reply:
Sounds good, lets go for it.